src/test/java/io/appium/java_client/android/IntentTest.java

```diff
@@ -47,26 +49,25 @@
     }
 
 
-    @Test public void startActivityWithIntent() {
-        final Activity activity = new Activity("com.android.mms", ".ui.ComposeMessageActivity");
-        activity.setIntentAction("android.intent.action.SEND");
-        activity.setIntentCategory("android.intent.category.DEFAULT");
-        activity.setIntentFlags("0x4000000");
-        activity.setOptionalIntentArguments("-d \"TestIntent\" -t \"text/plain\"");
-        driver.startActivity(activity);
-        try {
-            Thread.sleep(5000);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-        }
+    @Test public void startActivityWithIntent() throws Exception {
+        assertTrue(((Predicate<AndroidDriver>) driver -> {
+            final Activity activity = new Activity("com.android.mms", ".ui.ComposeMessageActivity");
+            activity.setIntentAction("android.intent.action.SEND");
+            activity.setIntentCategory("android.intent.category.DEFAULT");
+            activity.setIntentFlags("0x4000000");
+            activity.setOptionalIntentArguments("-d \"TestIntent\" -t \"text/plain\"");
+            driver.startActivity(activity);
+            return true;
```

Comment:
@mykola-mokhnach 
Yes. It is for codecy checkings. It requires assertions. Also I agree that assertions inside tests methods is the good style.